src/main/webapp/video_context.js

```diff
@@ -12,30 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-google.charts.load('current', {'packages': ['corechart']});
-google.setOnLoadCallback(updateUIWithVideoContext);
-
 /**
  * Retrieve the context of a Youtube video
+ * @param {String} urlInput inputted url of video
  * @return {Json} Json object of video context information
  */
-async function getVideoContext() {
-  const urlInput = document.getElementById('link-input');
-  const url = extractYouTubeUrl(urlInput.value);
+async function getVideoContext(urlInput) {
+  const url = cleanseUrl(urlInput);
   const response = await fetch('/VideoContext?url=' + url);
   const context = await response.json();
   return context;
 }
 
 /**
  * Fetches data and adds to html
+ * @param {string} url of the video being analyzed
  */
-async function updateUIWithVideoContext() {
-  $('form').submit(async function() {
-    videoContext = await getVideoContext();
-    document.getElementById('video-context').innerHTML =
+async function updateUIWithVideoContext(url) {
+  clearElement('video-context');
+  if (url == undefined) {
```

Comment:
As with my comment on `GetChart` can we update this method to always call it with a URL? That way information about whether a video was searched for via keyword search or by link doesn't leak to multiple places.